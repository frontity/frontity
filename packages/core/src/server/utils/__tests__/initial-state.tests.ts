import initialState from "../initial-state";

const settings = {
  name: "site",
  mode: "html",
  state: {
    frontity: {
      url: "https://site.com",
      prop1: "prop1",
      menu1: ["item1"]
    }
  },
  packages: [
    {
      name: "package1",
      active: true,
      state: {
        frontity: {
          prop1: "overwritten by package1",
          menu1: ["item2", "item3"]
        },
        package1: {
          prop2: "prop2"
        }
      }
    },
    {
      name: "package2",
      active: true,
      state: {
        frontity: {
          prop1: "overwritten by package2"
        },
        package2: {
          prop3: "prop2"
        }
      }
    }
  ]
};

describe("initialState", () => {
  it("should return a valid initial state", () => {
    const url = new URL("https://site.com/post");
    expect(initialState({ settings, url })).toMatchSnapshot();
  });

  it("should return a valid path and page", () => {
    const url = new URL("https://site.com/category/nature/page/2");
    expect(initialState({ settings, url })).toMatchSnapshot();
  });

  it("should return a valid path and page with home", () => {
    const url = new URL("https://site.com");
    expect(initialState({ settings, url })).toMatchSnapshot();
  });

  it("should return a valid path and page with paginated home", () => {
    const url = new URL("https://site.com/page/2");
    expect(initialState({ settings, url })).toMatchSnapshot();
  });
});
