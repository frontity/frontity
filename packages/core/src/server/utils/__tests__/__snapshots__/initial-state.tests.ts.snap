// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialState should return a valid initial state 1`] = `
Object {
  "frontity": Object {
    "initial": Object {
      "page": 1,
      "path": "/post",
    },
    "menu1": Array [
      "item1",
      "item2",
      "item3",
    ],
    "mode": "html",
    "name": "site",
    "packages": Array [
      "package1",
      "package2",
    ],
    "platform": "server",
    "prop1": "overwritten by package2",
    "url": "https://site.com",
  },
  "package1": Object {
    "prop2": "prop2",
  },
  "package2": Object {
    "prop3": "prop2",
  },
}
`;

exports[`initialState should return a valid path and page 1`] = `
Object {
  "frontity": Object {
    "initial": Object {
      "page": 2,
      "path": "/category/nature/",
    },
    "menu1": Array [
      "item1",
      "item2",
      "item3",
    ],
    "mode": "html",
    "name": "site",
    "packages": Array [
      "package1",
      "package2",
    ],
    "platform": "server",
    "prop1": "overwritten by package2",
    "url": "https://site.com",
  },
  "package1": Object {
    "prop2": "prop2",
  },
  "package2": Object {
    "prop3": "prop2",
  },
}
`;

exports[`initialState should return a valid path and page with home 1`] = `
Object {
  "frontity": Object {
    "initial": Object {
      "page": 1,
      "path": "/",
    },
    "menu1": Array [
      "item1",
      "item2",
      "item3",
    ],
    "mode": "html",
    "name": "site",
    "packages": Array [
      "package1",
      "package2",
    ],
    "platform": "server",
    "prop1": "overwritten by package2",
    "url": "https://site.com",
  },
  "package1": Object {
    "prop2": "prop2",
  },
  "package2": Object {
    "prop3": "prop2",
  },
}
`;

exports[`initialState should return a valid path and page with paginated home 1`] = `
Object {
  "frontity": Object {
    "initial": Object {
      "page": 2,
      "path": "/",
    },
    "menu1": Array [
      "item1",
      "item2",
      "item3",
    ],
    "mode": "html",
    "name": "site",
    "packages": Array [
      "package1",
      "package2",
    ],
    "platform": "server",
    "prop1": "overwritten by package2",
    "url": "https://site.com",
  },
  "package1": Object {
    "prop2": "prop2",
  },
  "package2": Object {
    "prop3": "prop2",
  },
}
`;
